firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/vector.txt

```diff
@@ -0,0 +1,60 @@
+Scenario: The function in Util Vectors works correctly
+  GIVEN("The util resize function")
+  WHEN("the util resize function is called on 252UL buffer size")
+    THEN("Result is as expected ")
```

Comment:
For non-`const` methods, we need to be more specific than "result is as expected". Specifically, we need to test:
- The value of the method's return
- The behavior of the `Vector` after the method is called, i.e. what happens when we call the same method or other methods afterwards.